### ISSUE TITLE
JAWS - Autocomplete doesn't announce dropdown closing for invalid input

### ISSUE THREAD

**Issue by DaveBest99** [angular/material2#3375]
#### Bug, feature request, or proposal:
Request: Typed input should only match what is in the list box.

#### What is the expected behavior?
In a non-editable auto-complete, any letters that do not result in a match from the list should be ignored, the drop down list of choices remains static until the user presses Escape to clear the text field, Backspace to remove some of the letters previously typed, or types an additional letter that results in a valid list of choices. In an editable auto-complete, if no choices match the letters typed, the drop down list closes. 

#### What is the current behavior?
Currently anything can be typed into the Edit box.

#### What is the use-case or motivation for changing an existing behavior?
Help users from making mistakes. A non-editable list box should only allow typed text that matches the list items, and an editable list box can accept typed text that does not match the list items, but this is confusing in a list of States.

#### Is there anything else we should know?
The only time the Edit box gets screen reader focus is when an invalid character is typed. That is, as long as there is a list item with the sequence of typed letters, the focus will move through the list, but a non-recognizable sequence will bring focus to the Edit box. Deleting the last letter typed (pressing Backspace) will move focus to the valid list item but it is not announced by the screen reader. That is, the screen reader user does not know what list item has focus (Note,although JAWS does not announce the State name, it is showned on the braille display). The user can press up/down arrow keys through the list but never get Edit box focus. Focus should stay on the Edit box until the arrow key is pressed, and an aria-alert can read the list item in focus.

**Comment by kara** [comment]
Related: Related: https://github.com/angular/material2/issues/3416

**Comment by kara** [comment]
Currently we hide the panel when there are no options, but it looks like aria-expanded is not updated properly, so there's no indication that the list is closed. That's definitely a bug. It's also weird when the invalid letter is deleted because it doesn't announce that the list has re-opened. Looking into it.

**Comment by DaveBest99** [comment]
I think Github3416 offers a cleaner solution for opening and closing the list.
